lib/puppet/face/strings.rb

```diff
@@ -50,6 +50,7 @@ def check_required_features
       # For now, assume the remaining positional args are a list of manifest
       # and ruby files to parse.
       yard_args = (args.empty? ? MODULE_SOURCEFILES : args)
+      class YARD::Logger; def progress(*args); end; end
```

Comment:
Actually this is for debugging purposes and should be removed entirely. It's gutting a method so that threads do not get in the way of debugging.